boofuzz/monitors/pedrpc.py

```diff
@@ -98,13 +100,25 @@ def __method_missing(self, method_name, *args, **kwargs):
         if method_name == "__bool__":
             return 1
 
+        # subclasses run into this as they call a trampoline method...
+        # not sure if this is the right way to handle it, but it seems to work
+        if method_name.endswith("__method_missing"):
+            return self.__method_missing(*args, **kwargs)
+        elif method_name.endswith("__hot_transmit"):
+            return self.__hot_transmit(*args, **kwargs)
+
```

Comment:
Right, this is one of my grievances with the PED-RPC protocol. It strictly assumes that one message is sent per connection, which doesn't work too well with restoring options when a new server connection is detected. _hot_transmit is a hack to call options on the remote RPC service which uses the current connection opened by the command that detected a new server, sets an option and resets the connection. This is done for every option that is set, so it is called a total of n times if n options are set, leaving the logic in a state where a connection is open (thus the "hot" part of _hot_transmit) to be then resumed normally with the call of the RPC function that was originally called.

Also, this logic shamelessly assumes that the RPC server is not restarted while restoring options. I think this is reasonable, because it would complicate things even more if we were to check this here as well. This is one of the main reasons why I -- if I had a business case -- would scrap the implementation and do something with RPyC instead as it is *much* easier to work with in the end.

Reply:
Thanks for the explanation!